central/complianceoperator/v2/report/manager/complianceReportgenerator/report_gen_impl.go

```diff
@@ -0,0 +1,314 @@
+type resultRow struct {
+	ClusterName string
+	CheckName   string
+	Profile     string
+	ControlRef  string
+	Description string
+	Status      string
+	Remediation string
+}
+
+type resultEmail struct {
+	resultCSVs map[string][]*resultRow //map of cluster id to slice of *resultRow
+	totalPass  int
+	totalFail  int
+	totalMixed int
+	profiles   []string
+	clusters   int
+}
+
+func (rg *complianceReportGeneratorImpl) ProcessReportRequest(ctx context.Context, req *ComplianceReportRequest) error {
+
+	data, err := rg.getDataforReport(req, ctx)
+	if err != nil {
+		return err
+	}
+
+	zipData, err := rg.Format(data.resultCSVs)
+	if err != nil {
+		return err
+	}
+	profiles := strings.Join(req.profiles, ", ")
+	formatEmailBody := &formatBody{
+		BrandedPrefix: branding.GetCombinedProductAndShortName(),
+		Profile:       profiles,
+		Pass:          data.totalPass,
+		Fail:          data.totalFail,
+		Mixed:         data.totalMixed,
+		Cluster:       len(req.clusterIDs),
+	}
+	formatEmailBody.ComplianceStatus = "non compliant"
+	if data.totalFail == 0 {
+		formatEmailBody.ComplianceStatus = "compliant"
+	}
+
+	if len(profiles) > maxNumberProfilesinSubject {
+		profiles = strings.Join(req.profiles[0:maxNumberProfilesinSubject], ", ")
+		profiles += "..."
+	}
+
+	formatEmailSub := &formatSubject{
+		BrandedPrefix: branding.GetCombinedProductAndShortName(),
+		ScanConfig:    req.scanConfigName,
+		Profiles:      profiles,
+	}
+
+	return rg.sendEmail(zipData, formatEmailBody, formatEmailSub, req.notifiers, ctx)
+}
+
+// getDataforReport returns map of cluster id and
+func (rg *complianceReportGeneratorImpl) getDataforReport(req *ComplianceReportRequest, ctx context.Context) (*resultEmail, error) {
+	clusters := req.clusterIDs
+
+	var resultsCSV map[string][]*resultRow
+
+	resultEmailComplianceReport := &resultEmail{
+		totalPass:  0,
+		totalMixed: 0,
+		totalFail:  0,
+		clusters:   0,
+	}
+
+	for _, clusterID := range clusters {
+
+		parsedQuery := search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorScanConfig, req.scanConfigID).
+			AddExactMatches(search.ClusterID, clusterID).
+			ProtoQuery()
+		resultCluster := []*resultRow{}
+
+		err := rg.checkResultsDS.WalkByQuery(ctx, parsedQuery, func(c *storage.ComplianceOperatorCheckResultV2) error {
+			// TODO: Add query to get profile name
+			row := &resultRow{
+				ClusterName: c.GetClusterName(),
+				CheckName:   c.GetCheckName(),
+				Description: c.GetDescription(),
+				Status:      c.GetStatus().String(),
+				Remediation: c.GetInstructions(),
+			}
+			//get profile name for the check result
+			q := search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorRuleRef, c.GetRuleRefId()).ProtoQuery()
+			rule, err := rg.rulesDS.SearchRules(ctx, q)
+			if err != nil {
+				return err
+			}
+			if len(rule) < 1 {
+				return errors.New("Rule not found")
+			}
+			q = search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorRuleName, rule[0].GetName()).ProtoQuery()
```

Comment:
@ksurabhi91 Try this.  We can use the `ScanRefID` in the result.  That will join `ComplianceOperatorScanV2` and `ComplianceOperatorProfileV2`.   That should give you the profile you are looking for as that is a 1:1 mapping between those tables.

```suggestion
			//get profile name for the check result
			q := search.NewQueryBuilder().
		AddExactMatches(search.ComplianceOperatorScanRef, c.GetScanRefID()).ProtoQuery()
```